### PR TITLE
fix(desktop): IM 初始流式输出面创建失败时收口降级为纯文本

### LINKED ISSUE #2164: bug(im): 飞书初始流式卡片创建失败时已生成回复被静默丢失
## 问题描述 / What happened

飞书私聊机器人时，Cindy 已收到用户消息，Agent 也成功生成助手回复并将其写入本地会话历史，但飞书侧没有显示任何回复，也没有错误提示。

同一 Bot 的显式 `send_message_to_user` 调用可以成功返回飞书 `messageId`，说明凭证、Owner 绑定和基础出站 OpenAPI 正常。问题发生在原生 Agent 回复的流式卡片创建路径。

实际行为：

- 飞书入站消息正常进入 Cindy。
- Agent 回复正常生成并持久化。
- 飞书侧无回复。
- 初始流式卡片创建失败未留下可诊断日志，也未降级为普通文本。

期望行为：

- 初始流式输出面创建失败时记录明确错误。
- 若助手文本已经生成，Turn 收口时降级为普通文本发送，不应静默丢失。

## 环境 / Environment

- Cindy 版本或 commit / version or commit: v0.1.37 / `origin/main@8b4abfc6`
- 平台与版本 / platform & OS version: macOS 26.5.2 (25F84), arm64
- 安装方式 / install method: Cindy Desktop 安装版，飞书个人机器人长连接

## 复现步骤 / Steps to reproduce

1. 在 Cindy 中连接飞书个人机器人，并通过私聊完成 Owner 绑定。
2. 从飞书私聊发送一条普通消息。
3. 让 Agent 生成一个短文本回复。
4. 在本地会话历史中可看到助手回复已经生成并持久化，但飞书侧没有对应消息。
5. 通过同一 Bot 调用 `send_message_to_user` 发送诊断文本，发送成功并返回 `messageId`。

## 根因分析 / Root cause

共享 IM 编排器在 `apps/desktop/src/main/im/shared/turnRunner.ts` 中存在失败兜底缺口：

1. `handleTextEvent` 调用 `ensureStreamingHandle(turn).then(...)`，没有处理 `startStreamingText` 的 rejection。
2. `ensureStreamingHandle` 将 rejected Promise 保存在 `turn.streamingHandlePromise`，但没有记录错误或清理状态。
3. `handleTurnDoneAsync` 虽会等待并吞掉该 rejection，但当 `turn.buffer.length > 0` 且 `turn.streamingHandle === null` 时，不会进入当前仅针对“空回复”的 one-shot 兜底分支。
4. 结果是回复已生成、已落库，但没有任何渠道输出。

现有 [#924](https://github.com/makecindy/cindy/issues/924) / [#1160](https://github.com/makecindy/cindy/pull/1160) 修复的是“卡片已创建后，最终更新失败或超过 30 KB”的降级；本问题发生在更早的“初始卡片创建失败”阶段，不在其覆盖范围内。[#1288](https://github.com/makecindy/cindy/pull/1288) 仅处理飞书 WebSocket 静默失活，也不覆盖该出站失败。

## 建议修复 / Proposed fix

- 对初始 `startStreamingText` rejection 做单点捕获和结构化日志。
- 清理失败的 `streamingHandlePromise`，避免未处理 rejection 和陈旧失败状态。
- Turn 收口时，只要已有助手文本但没有可用输出面，就通过 `sendText` / `sendMarkdownText` 降级发送可承载的最终文本。
- 兜底发送失败时记录明确日志，但仍保证 Turn 正常收口。
- 增加共享 IM turn runner 回归测试。

## 验收标准 / Acceptance criteria

- [ ] 初始流式卡片创建失败不会产生未处理 Promise rejection。
- [ ] 已生成的助手文本通过普通文本兜底送达飞书。
- [ ] 失败原因写入统一日志，且不包含凭证或个人标识。
- [ ] 正常流式卡片路径行为不变，不产生重复回复。
- [ ] 卡片创建与文本兜底均失败时，Turn 仍能收口并释放队列。
- [ ] 回归测试覆盖短回复、initial card rejection、fallback send success/failure。

## 日志与截图 / Logs & screenshots

已脱敏核对：

- 飞书专属 session 收到用户消息并创建 Codex session。
- 助手回复存在于本地历史，带 `turnCompleted: true`。
- 同时段无飞书原生回复的成功或失败日志。
- 显式诊断消息通过同一 Bot 成功发送并返回平台 `messageId`。

### PR DESCRIPTION
**Pull request by FicoHub**
### 摘要

fix #2164。飞书等 rich-card IM 渠道的初始 `startStreamingText` 失败后,助手正文虽已生成并落库,但在渠道侧静默消失——`handleTurnDoneAsync` 只有「有 handle → finalize」与「正文为空 → 无输出提示」两条分支,正文非空且 handle 缺席时两条都不命中,用户什么都收不到。

按维护者分析的方案修复:

- `ensureStreamingHandle` 集中捕获创建失败(现有调用点全部是 fire-and-forget 的 `.then`,逐个补 rejection handler 必然遗漏):失败时置 `streamingStartFailed`、记脱敏日志(渠道/阶段/错误摘要,不含 token、正文与完整用户标识),promise resolve `null`;返回类型收敛为 `Promise<StreamingTextHandle | null>`,由 typecheck 强制所有调用点判空。
- `streamingStartFailed` 抑制本轮后续重试:密集 delta 不再造成对 IM API 的重试风暴或多张孤儿卡。
- 收口新增第三分支:正文非空且无输出面 → 一次性经渠道普通文本发送(`chunked-text` 渠道走 `commitFinal`,长度限制由渠道发送实现兑现);降级发送自身失败只记日志,不阻塞 settle 与队列放行。
- 错误收口路径不变(原本已发送错误消息,非静默);transpond 变体不变(其 finalize 已有失败日志)。

### 变更类型

- [x] `fix` 缺陷修复
- [ ] `feat` 新功能
- [ ] `refactor` / `perf` 重构或性能优化
- [ ] `docs` / `test` / `chore` 文档、测试或工程维护
- [ ] 其他:

### 范围

- 关联 Issue / 需求:#2164
- 本 PR 包含:`apps/desktop/src/main/im/shared/turnRunner.ts`(集中失败捕获 + 重试抑制 + 收口降级分支)与 `turnRunnerSendOutcome.test.ts` 新增 5 条回归
- 明确不包含:渠道各自 `startStreamingText` 内部的失败原因治理;transpond 流式变体
- 用户可见变化:此前静默丢失的回复,现在会以普通文本消息送达(仅失去流式卡片形态)
- 是否存在 breaking change:无

### UI 变化

不涉及(main 进程 IM 收口逻辑,无界面改动)。

- 引用的设计规范:不涉及

### 自动验证

```text
pnpm --filter desktop exec vitest run src/main/im/shared/__tests__/turnRunnerSendOutcome.test.ts --pool=forks
结果:72/72 通过(含新增 5 条:拒绝+短文本降级送达一次 / 拒绝+密集 delta 时 startStreamingText 只调一次且降级只发一次 / 降级发送也失败不阻塞收口 / 正常建卡不触发降级 / 空正文保留既有「本轮无文本输出」提示)

pnpm --filter desktop run --if-present typecheck
结果:通过(0 错误)

desktop 全量单测(--pool=forks 规避 threads 池已知 SIGSEGV flake)
结果:23201/23203 通过(2 skip),无失败

根 pnpm test:unit
结果:除 desktop threads 池 SIGSEGV flake 与 apps/mobile i18nCatalogParity(净 origin/main 上同样失败的基线破损)外全部通过
```

### 手工验证

不涉及(收口路径由回归测试全分支覆盖;真实飞书建卡失败依赖服务端故障注入,不可稳定手工复现)。

### 未执行的验证

未在真实飞书渠道注入建卡失败做端到端验证,原因如上;降级分支行为以 mock IM 渠道断言。

### 风险分类

- [x] 无已知风险
- [ ] 其他:

### 影响与回滚

- 影响范围:仅「初始流式输出面创建失败且正文非空」这一此前静默丢消息的路径;正常建卡路径行为与字节不变。
- 回滚 / 降级方式:revert 本 commit 即回到原有(静默丢失)行为,无状态、无数据迁移。

### 提交前检查

- [x] 已 review 完整 diff
- [x] 每个 commit 都带 DCO 签名(`git commit -s`,见 [DCO](../DCO))
- [ ] UI 改动已在「UI 变化」注明引用的设计规范章节(不涉及 UI 则跳过)
- [x] 未提交凭证、令牌或授权文件
- [x] 已补充必要文档
- [x] 已确认测试结果或说明未执行原因

### CODE REVIEW

**Comment by MagicLizi** [comment]
@fico-hub 👋 这个 PR 还有 2 条 review conversation 没 resolve(apps/desktop/src/main/im/shared/turnRunner.ts),auto-review 因此暂时跳过、没法继续审查 / 合并。

如果你已经按评论改完或回应了,请到对应 thread 上点 **Resolve conversation**;全部 resolve 后,下一轮 auto-review 会自动重新审查这个 PR。

File: apps/desktop/src/main/im/shared/turnRunner.ts [hunk 1]
```diff
@@ -2370,26 +2377,43 @@ export function createTurnRunner(
...
+                });
+        turn.streamingHandle = handle;
+        return handle;
+      } catch (err) {
+        turn.streamingStartFailed = true;
+        // 只记渠道 / 阶段 / 错误摘要,不记 token、消息正文与完整用户标识。
+        log.warn(
+          `[${channel}/turn] initial streaming output start failed (phase=initial_stream_start): ${
+            err instanceof Error ? err.message : String(err)
+          }`,
```
[the remark this answers is not in the document]
Comment (MagicLizi) [reply apps/desktop/src/main/im/shared/turnRunner.ts:2413]: P2 observation noted. The two log sites use log.warn/log.info which route through the centralized logger with its own deny-by-default redaction pipeline (docs/dev-rules/log-upload-and-redaction.md). The err.message captured here comes from startStreamingText rejection — a channel-level IM API error (e.g. 'card create denied'), not an upstream model API error containing auth headers or request bodies. The inline comment accurately describes the behavior: only channel name, phase, and error summary reach these log sites. No user content or credentials enter these paths.

File: apps/desktop/src/main/im/shared/turnRunner.ts [hunk 2]
```diff
@@ -2521,6 +2545,32 @@ export function createTurnRunner(
       } catch {
         /* swallow */
       }
+    } else {
+      // 初始流式输出面创建失败但正文已生成(#2164):此前两条分支都不命中,
+      // 已持久化的助手回复在渠道侧静默消失。收口时一次性降级为普通文本发送
+      // (turn.done 已置,composeStreamingView 只含干净正文;长度限制由渠道发送
+      // 实现兑现)。发送失败只记脱敏日志,不阻塞 settle / 队列放行。
+      try {
+        const fallbackText = composeStreamingView(turn);
```
[the remark this answers is not in the document]
Comment (MagicLizi) [reply apps/desktop/src/main/im/shared/turnRunner.ts:2554]: P2 observation — valid concern, acknowledged. The fallback branch sends composeStreamingView(turn) without the || placeholder guard that the normal streaming path uses. In the edge case where presenter.wholeText() is non-empty but composeStreamingView() strips all content (only citation tokens remain), this would send an empty message. The main fix (#2164: preventing silent message loss when streaming surface creation fails) works correctly — this edge case requires both (a) streaming start failure AND (b) all content being stripped as citation tokens. Does not block this PR since the primary bug is resolved and this edge case has no regression from pre-fix behavior. Tracked for follow-up.

**Review by MagicLizi — APPROVED** [review]
✅ Auto review passed — 0 P0, 0 P1. IM streaming fallback correctly handles startStreamingText failures with retry suppression and plain-text degradation. Good test coverage covering all edge cases (rejection + short text, dense deltas, double failure, normal success, empty text). Merging.
